scalyr_agent/monitor_utils/k8s.py

```diff
@@ -2142,34 +2147,63 @@ def __init__(self, k8s, port=10255, host_ip=None):
         self._session = requests.Session()
         headers = {
             "Accept": "application/json",
+            "Authorization": "Bearer %s" % k8s.token,
         }
         self._session.headers.update(headers)
 
-        global_log.info("KubeletApi host ip = %s" % host_ip)
-        if host_ip:
-            self._http_host = "http://%s:%d" % (host_ip, port)
-        else:
-            self._http_host = None
+        global_log.info("KubeletApi host ip = %s" % self._host_ip)
+        self._build_host()
```

Comment:
Let's declare two variables here, called `_kubelet_url` and `_fallback_kubelet_url`.  Let's set them by doing something like:
```
    self._kubelet_url = self._build_kubelet_url(url_template, host_ip)
    self._fallback_kubelet_url = self._build_kubelet_url(FALLBACK_KUBELET_URL_TEMPLATE, host_ip)
```

(where `FALLBACK_KUBELET_URL_TEMPLATE` is a constant variable defined in this file.)

The `_kubelet_url` will represent the URL we are currently using to contact the kubelet api.